src/main/scala/epic/dense/LowRankQuadraticTransform.scala

```diff
@@ -0,0 +1,142 @@
+  }
+
+  case class OutputLayer(sublayers: Seq[LRQTNLayer], innerLayer: innerTransform.Layer) extends OutputTransform.OutputLayer[FV,DenseVector[Double]] {
+    
+    override val index = LowRankQuadraticTransform.this.index
+    val neuronIndex = LowRankQuadraticTransform.this.neuronIndex
+
+    def activations(fv: FV) = {
+      val innerActivations = innerLayer.activations(fv)
+      DenseVector(Array.tabulate(sublayers.size)(i => sublayers(i).activations(innerActivations)(0)))
+    }
+    
+    def activationsFromPenultimateDot(innerLayerActivations: DenseVector[Double], sparseIdx: Int): Double = {
+      sublayers(sparseIdx).activations(innerLayerActivations)(0)
+    }
+
+    def tallyDerivative(deriv: DenseVector[Double], _scale: =>Vector[Double], fv: FV) = {
+      val innerActivations = innerLayer.activations(fv)
+      for (i <- 0 until sublayers.size) {
+        sublayers(i).tallyDerivative(deriv(neuronIndex.componentOffset(i) until neuronIndex.componentOffset(i) + neuronIndex.indices(i).size), _scale(i), innerActivations)
+      }
+    }
+    
+    def applyBatchNormalization(inputs: scala.collection.GenTraversable[FV]) = innerLayer.applyBatchNormalization(inputs)
+  }
+
+ 
+}
+
+/**
+ * Separate because I was having some issues...
+ */
+case class LowRankQuadraticTransformNeuron(numRanks: Int, numLeftInputs: Int, numRightInputs: Int) {
+
+  val index = SegmentedIndex(new AffineTransform.Index(numRanks, numLeftInputs, false), new AffineTransform.Index(numRanks, numRightInputs, false))
+
+  def extractLayer(weights: DenseVector[Double]) = {
+    val lhsSize = numRanks * numLeftInputs
+    val rhsSize = numRanks * numRightInputs
+    val lhsMat = weights(0 until lhsSize).asDenseMatrix.reshape(numRanks, numLeftInputs, view = View.Require)
+    val rhsMat = weights(lhsSize until (lhsSize + rhsSize)).asDenseMatrix.reshape(numRanks, numRightInputs, view = View.Require)
+    new LRQTNLayer(lhsMat, rhsMat, index, numRanks, numLeftInputs, numRightInputs)
+  }
+  
+  def initialWeightVector(initWeightsScale: Double, rng: Random, outputLayer: Boolean, spec: String) = {
+    if (spec == "magic") {
+      DenseVector.vertcat(AffineTransform.getMagicAffineWeights(index.indices(0).size, numLeftInputs, numRanks, initWeightsScale, rng),
+                          AffineTransform.getMagicAffineWeights(index.indices(1).size, numRightInputs, numRanks, initWeightsScale, rng))
+    } else {
+      DenseVector.vertcat(AffineTransform.getGaussianAffineWeights(index.indices(0).size, initWeightsScale, rng),
+                          AffineTransform.getGaussianAffineWeights(index.indices(1).size, initWeightsScale, rng))
+    } 
+  }
+  
+  def clipHiddenWeightVectors(weights: DenseVector[Double], norm: Double, outputLayer: Boolean) {
+  }
+}
+
+
+case class LRQTNLayer(lhsWeights: DenseMatrix[Double], rhsWeights: DenseMatrix[Double], index: Index[Feature], numRanks: Int, numLeftInputs: Int, numRightInputs: Int) {
+  val lhsWeightst = lhsWeights.t
```

Comment:
I forget if the copy was actually slowing things down or if I was just scared it was, but I'm pretty sure I tried it both ways.